plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1064,8 +1066,7 @@ public void setTableComment(ConnectorSession session, ConnectorTableHandle table
         DeltaLakeTableHandle handle = checkValidTableHandle(tableHandle);
         checkSupportedWriterVersion(session, handle.getSchemaTableName());
         ColumnMappingMode columnMappingMode = getColumnMappingMode(handle.getMetadataEntry());
-        if (columnMappingMode != NONE) {
-            // TODO https://github.com/trinodb/trino/issues/12638 Support setting a table comment for id and name column mapping mode
+        if (columnMappingMode != ID && columnMappingMode != NAME && columnMappingMode != NONE) {
```

Comment:
would be okay to just check `columnMappingMode == UNKNOWN`?

Reply:
No, the logic doesn't work if Delta Lake introduces a new column mapping mode.